unconvert.go

```diff
@@ -100,26 +102,41 @@ func (e *editor) rewrite(f *ast.Expr) {
 	delete(e.edits, pos)
 }
 
-func print(name string, edits editSet) {
-	if len(edits) == 0 {
-		return
-	}
+func print(edits []token.Position) {
+	fileByName := make(map[string][]byte)
 
-	buf, err := ioutil.ReadFile(name)
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	for pos := range edits {
+	for _, pos := range edits {
 		fmt.Printf("%s:%d:%d: unnecessary conversion\n", pos.Filename, pos.Line, pos.Column)
+
 		if *flagV {
-			line := string(lineForOffset(buf, pos.Offset))
-			fmt.Println(line)
-			fmt.Println(strings.Repeat(" ", stringWidth(line[:pos.Column-1])) + "^")
+			buf, ok := fileByName[pos.Filename]
+			if !ok {
+				var err error
+				buf, err = ioutil.ReadFile(pos.Filename)
+				if err != nil {
+					log.Fatal(err)
+				}
+				fileByName[pos.Filename] = buf
+			}
+
+			line := getLine(bytes.NewReader(buf), pos.Line)
```

Comment:
One thing to be mindful of is that some .go files have really large lines, longer than 64 KB, and something inside `bufio` uses https://godoc.org/bufio#MaxScanTokenSize (I forget if it's `Reader` or `Scanner` off the top of my head).